### PR TITLE
Add folder structure

### PR DESCRIPTION
**Pull request by ZlataKutuzava**

### CODE REVIEW

File: hw21/pageObjects/basePage.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  }
+
+  public async clickOnLocationSelectionButton() {
+    return await (
+      await this.driverUtils.findElement(
+        SELECTOR_TYPES.CSS,
+        "button.location-selector__button"
+      )
+    ).click();
+  }
```
Comment (VladSelikh) [inline hw21/pageObjects/basePage.ts:71]: Please correct:
```
public async clickOnLocationSelectionButton() {
    await (await this.getLocationSelectionPanel()).click();
  }
```

File: hw21/pageObjects/contactUsPage.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+  public async getFirstNameField() {
+    return await this.driverUtils.findElement(
+      SELECTOR_TYPES.CSS,
+      "input[name='user_first_name']"
+    );
+  }
+
+  public async enterFirstName(firstName: string) {
+    return (await this.getFirstNameField()).sendKeys(firstName);
```
Comment (VladSelikh) [inline hw21/pageObjects/contactUsPage.ts:18]: There is no need to return this statement. Please delete `return`.

File: hw21/pageObjects/contactUsPage.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+  public async getEmailField() {
+    return await this.driverUtils.findElement(
+      SELECTOR_TYPES.CSS,
+      "input[name='user_email']"
+    );
+  }
+
+  public async enterEmail(email: string) {
+    return (await this.getEmailField()).sendKeys(email);
```
Comment (VladSelikh) [inline hw21/pageObjects/contactUsPage.ts:40]: There is no need to return this statement. Please delete `return`.

File: hw21/pageObjects/contactUsPage.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+  public async getPhoneField() {
+    return await this.driverUtils.findElement(
+      SELECTOR_TYPES.CSS,
+      "input[name='user_phone']"
+    );
+  }
+
+  public async enterPhone(phone: string) {
+    return (await this.getPhoneField()).sendKeys(phone);
```
Comment (VladSelikh) [inline hw21/pageObjects/contactUsPage.ts:51]: There is no need to return this statement. Please delete `return`.

File: hw21/pageObjects/contactUsPage.ts [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+  }
+
+  public async submitForm() {
+    return await (
+      await this.driverUtils.findElement(
+        SELECTOR_TYPES.CSS,
+        "button[type='submit']"
+      )
+    ).click();
+  }
```
Comment (VladSelikh) [inline hw21/pageObjects/contactUsPage.ts:61]: There is no need to return this statement. Please delete `return`.

File: hw21/tests/test.spec.ts [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+  });
+
+  it("Our Work page contains 11 links", async () => {
+    await homePage.clickOnNavigationItemByText(NAVIGATION_ITEMS.OUR_WORK);
+    expect(
+      await (
+        await ourWork.getPageHeader(SELECTOR_TYPES.XPATH, "//h1")
+      ).getText()
+    ).to.equal(PAGES.OUR_WORK);
+    expect(await ourWork.getHorizontalMenuItem()).to.have.length(11);
```
Comment (VladSelikh) [inline hw21/tests/test.spec.ts:82]: Please create a constant for links amount and use it in the test name also with the help of string template.

File: hw21/tests/test.spec.ts [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+    ).not.to.contain("opened");
+  });
+
+  it("Warning message in displayed on the Contact Page when user submits form with the incorrect phone", async () => {
+    await homePage.clickContactUsButton();
+    expect(
+      await (
+        await contactUs.getPageHeader(SELECTOR_TYPES.XPATH, "//h1")
+      ).getText()
+    ).to.equal("Contact Us");
```
Comment (VladSelikh) [inline hw21/tests/test.spec.ts:60]: Your already hame the correcponding enum for this text, please use it.

File: hw21/pageObjects/basePage.ts [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+
+  public async quitBrowser() {
+    await this.driver.quit();
+  }
+
+  public async waitForTitleToContain(title: string) {
+    await this.driver.wait(until.titleContains(title));
+  }
+
+  public async getPageHeader(
```
Comment (VladSelikh) [inline hw21/pageObjects/basePage.ts:28]: Get page header by what?

Comment (ZlataKutuzava) [reply hw21/pageObjects/basePage.ts:28]: Edited method as getPageHeaderByXpath, as in all scenarios xpath selector type is used

File: hw21/pageObjects/contactUsPage.ts [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+
+  public async getPhoneField() {
+    return await this.driverUtils.findElement(
+      SELECTOR_TYPES.CSS,
+      "input[name='user_phone']"
+    );
+  }
+
+  public async enterPhone(phone: string) {
+    (await this.getPhoneField()).sendKeys(phone);
```
Comment (VladSelikh) [inline hw21/pageObjects/contactUsPage.ts:51]: Please use:
```
await (await this.getPhoneField()).sendKeys(phone);
```

File: hw21/pageObjects/contactUsPage.ts [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+
+  public async getEmailField() {
+    return await this.driverUtils.findElement(
+      SELECTOR_TYPES.CSS,
+      "input[name='user_email']"
+    );
+  }
+
+  public async enterEmail(email: string) {
+    (await this.getEmailField()).sendKeys(email);
```
Comment (VladSelikh) [inline hw21/pageObjects/contactUsPage.ts:40]: Please use:
```
await (await this.getEmailField()).sendKeys(email);
```

File: hw21/pageObjects/contactUsPage.ts [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+
+  public async getLastNameField() {
+    return await this.driverUtils.findElement(
+      SELECTOR_TYPES.CSS,
+      "input[name='user_last_name']"
+    );
+  }
+
+  public async enterLastName(lastName: string) {
+    (await this.getLastNameField()).sendKeys(lastName);
```
Comment (VladSelikh) [inline hw21/pageObjects/contactUsPage.ts:29]: Please use:
```
await (await this.getLastNameField()).sendKeys(lastName);
```

File: hw21/pageObjects/basePage.ts [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+  public async clickOnNavigationItemByText(item: NAVIGATION_ITEMS) {
+    await (await this.getNavigationItemByText(item)).click();
+  }
+
+  public async clickOnLocationSelectionButton() {
+     await (await this.getLocationSelectionPanel()).click();
+  }
+
+  public async getLocationSelectionPanel() {
+    return await await this.driverUtils.findElement(
```
Comment (VladSelikh) [inline hw21/pageObjects/basePage.ts:68]: Please be attentive. You wrote `await` keyword twice.

File: hw21/utils/types.ts [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+  ID = "id",
+}
+
+export enum PAGES {
+  HOME = "HomePage",
+  CONTACT_US = "Contact Us",
+  OUR_WORK = "Our Work",
+}
+
+export const LINKS_AMOUNT = 11;
```
Comment (VladSelikh) [inline hw21/utils/types.ts:23]: There is no need to move your constant in the separate file. Please place it in the beginning of the test file (camelCase notation).
P.S. A little remark: please do not place your constants in the file as the types. They should be placed separately.

File: hw21/tests/test.spec.ts [hunk 14]
```diff
@@ -0,0 +1,90 @@
...
+
+  it("Life Sciences page can be opened by clicking on the link on the Our Work page", async () => {
+    await ourWork.clickOnOurPageMenuItem("Life Sciences");
+    expect(await driver.getCurrentUrl()).to.equal(
+      `${baseURL}our-work/life-sciences`
+    );
+  });
+
+  after(async () => {
+    homePage.quitBrowser();
```
Comment (VladSelikh) [inline hw21/tests/test.spec.ts:88]: You've forgot `await`.

**Review by VladSelikh — APPROVED** [review]

**Pull request merged**